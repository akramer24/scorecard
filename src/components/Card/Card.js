import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Divider from '../Divider/Divider';

/**
 * Rectangular container, used to display content.
 */

const Card = ({
  bordered,
  children,
  className,
  footer,
  footerStyle,
  header,
  headerStyle,
  id,
  padded,
  style,
}) => {
  return (
    <div
      className={classNames(
        'ari-ui-card',
        {
          'ari-ui-card-bordered': bordered,
          [className]: className
        }
      )}
      id={id}
      style={style}
    >
      {
        header
          ? (
            <div className="ari-ui-card-header" style={headerStyle}>
              <div className="ari-ui-small-padding">
                {header}
              </div>
              <Divider />
            </div>
          )
          : null
      }
      <div
        className={classNames(
          'ari-ui-card-content',
          {
            'ari-ui-small-padding': padded,
          }
        )}
      >
        {children}
      </div>
      {
        footer
          ? (
            <div className="ari-ui-card-footer" style={footerStyle}>
              <Divider />
              <div className="ari-ui-small-padding">
                {footer}
              </div>
            </div>
          )
          : null
      }
    </div>
  )
}

Card.defaultProps = {
  bordered: true,
  padded: true
}

Card.propTypes = {
  /**
   * Determines whether or not card has a border.
   */
  bordered: PropTypes.bool,
  /**
   * CSS class(es) applied to Card.
   */
  className: PropTypes.string,
  /**
   * Card footer.
   */
  footer: PropTypes.node,
  /**
   * CSS styling for footer.
   */
  footerStyle: PropTypes.object,
  /**
   * Card header.
   */
  header: PropTypes.node,
  /**
   * CSS styling for header.
   */
  headerStyle: PropTypes.object,
  /**
   * CSS id applied to Card.
   */
  id: PropTypes.string,
  /**
   * Determines whether or not card content is padded.
   */
  padded: PropTypes.bool,
  /**
   * CSS styling for card.
   */
  style: PropTypes.object
}

export default Card;