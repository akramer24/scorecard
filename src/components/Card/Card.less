.ari-ui-card {
  min-width: 150px;
  height: fit-content;
  margin: 10px;
}

.ari-ui-card-bordered {
  box-shadow: 0px 0px 1px 1px @secondary-color;
}

.ari-ui-small-padding {
  padding: 5px;
}