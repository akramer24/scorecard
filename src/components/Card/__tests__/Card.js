import React from 'react';
import ReactDOM from 'react-dom';
import { configure, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Card from '../Card';

configure({ adapter: new Adapter() });

const mockProps = {
  bordered: false,
  children: 'Test Card',
  className: 'test-card',
  footer: <div>Footer</div>,
  footerStyle: { backgroundColor: 'blue' },
  header: <div>Header</div>,
  headerStyle: { backgroundColor: 'gray' },
  id: 'test-card-id',
  padded: true,
  style: { backgroundColor: 'green' },
}

describe('Card', () => {
  let tree;
  let wrapper;

  beforeAll(() => {
    tree = render(<Card {...mockProps} />);
    wrapper = mount(<Card {...mockProps} />);
  })

  afterAll(() => {
    wrapper.unmount();
  });

  beforeEach(() => {
    wrapper.setProps(mockProps);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card {...mockProps} />, div);
  });

  it('renders expected Jest snapshot correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('renders the data inside it', () => {
    expect(wrapper.text()).toEqual('HeaderTest CardFooter');
  });

  it('has the CSS class "ari-ui-card"', () => {
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass('ari-ui-card')
    ).toEqual(true);
  });

  it('has the CSS class "ari-ui-card-bordered" if bordered=true', () => {
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass('ari-ui-card-bordered')
    ).toEqual(false);
    wrapper.setProps({ bordered: true });
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass('ari-ui-card-bordered')
    ).toEqual(true);
  });

  it('has the CSS class passed via props as className', () => {
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass(mockProps.className)
    ).toEqual(true);
    wrapper.setProps({ className: '' });
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass(mockProps.className)
    ).toEqual(false);
  });

  it('renders a header if one is passed via props', () => {
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-header`)
        .length
    ).toEqual(1);
    wrapper.setProps({ header: null });
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-header`)
        .length
    ).toEqual(0);
  });

  it('passes headerStyle, if defined via props, to header', () => {
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-header`)
        .getDOMNode()
        .style
        ._values['background-color']
    ).toEqual('gray');
  });

  it('has a content div with CSS class "ari-ui-small-padding" if padded=true', () => {
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-content`)
        .hasClass('ari-ui-small-padding')
    ).toEqual(true);
    wrapper.setProps({ padded: false });
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-content`)
        .hasClass('ari-ui-small-padding')
    ).toEqual(false);
  });

  it('renders a footer if one is passed via props', () => {
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-footer`)
        .length
    ).toEqual(1);
    wrapper.setProps({ footer: null });
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-footer`)
        .length
    ).toEqual(0);
  });

  it('passes footerStyle, if defined via props, to footer', () => {
    expect(
      wrapper
        .find(`#${mockProps.id} .ari-ui-card-footer`)
        .getDOMNode()
        .style
        ._values['background-color']
    ).toEqual('blue');
  });
});