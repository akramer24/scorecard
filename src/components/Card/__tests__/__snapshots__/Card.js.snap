// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders expected Jest snapshot correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ari-ui-card test-card",
      "id": "test-card-id",
      "style": "background-color:green",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ari-ui-card-header",
          "style": "background-color:gray",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ari-ui-small-padding",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Header",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "ari-ui-divider ari-ui-divider-horizontal",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "ari-ui-divider ari-ui-divider-horizontal",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "ari-ui-small-padding",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Header",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ari-ui-card-content ari-ui-small-padding",
          },
          "children": Array [
            Object {
              "data": "Test Card",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "ari-ui-card-footer",
              "style": "background-color:blue",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ari-ui-divider ari-ui-divider-horizontal",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "ari-ui-small-padding",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Footer",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {},
                      "x-attribsPrefix": Object {},
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "ari-ui-small-padding",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Footer",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "ari-ui-divider ari-ui-divider-horizontal",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ari-ui-card-content ari-ui-small-padding",
        },
        "children": Array [
          Object {
            "data": "Test Card",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ari-ui-card-footer",
            "style": "background-color:blue",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ari-ui-divider ari-ui-divider-horizontal",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "ari-ui-small-padding",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Footer",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "ari-ui-small-padding",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Footer",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "ari-ui-divider ari-ui-divider-horizontal",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ari-ui-card-header",
            "style": "background-color:gray",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ari-ui-small-padding",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Header",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "ari-ui-divider ari-ui-divider-horizontal",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "ari-ui-divider ari-ui-divider-horizontal",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "ari-ui-small-padding",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Header",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ari-ui-card-footer",
          "style": "background-color:blue",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ari-ui-divider ari-ui-divider-horizontal",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "ari-ui-small-padding",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Footer",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "ari-ui-small-padding",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Footer",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "ari-ui-divider ari-ui-divider-horizontal",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ari-ui-card-content ari-ui-small-padding",
          },
          "children": Array [
            Object {
              "data": "Test Card",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "ari-ui-card-header",
              "style": "background-color:gray",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ari-ui-small-padding",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Header",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "ari-ui-divider ari-ui-divider-horizontal",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "ari-ui-divider ari-ui-divider-horizontal",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "ari-ui-small-padding",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "Header",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {},
                      "x-attribsPrefix": Object {},
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "id": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "id": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
